### PR TITLE
Siproxd plugin

### PR DESCRIPTION
**Pull request by mimugmail**
Siproxd plugin as requested in #246 

In #246 are some screenshots how it looks like. 

Current problem are the "Current registrations", the newlines are not correctly formatted. Perhaps you have some ideas? I could als try to write a script but a oneliner would be smarter I think.

setup.sh needs a chmod 755 as always when I contribute :)

### CODE REVIEW

File: net/siproxd/src/opnsense/mvc/app/models/OPNsense/Siproxd/ACL/ACL.xml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<acl>
+   <page-services-clamav>
+      <name>Services: ClamAV</name>
+      <patterns>
+         <pattern>ui/clamav/*</pattern>
+         <pattern>api/clamav/*</pattern>
```
Comment (fabianfrz) [inline net/siproxd/src/opnsense/mvc/app/models/OPNsense/Siproxd/ACL/ACL.xml]: clamav?

File: net/siproxd/src/opnsense/mvc/app/views/OPNsense/Siproxd/general.volt [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+                    <td colspan="5"></td>
+                <td>
+                    <button data-action="add" type="button" class="btn btn-xs btn-default"><span class="fa fa-plus"></span></button>
+                    </td>
+                </tr>
+            </tfoot>
+        </table>
+    </div>    
+    <div id="showregistrations" class="tab-pane fade in">
+      <pre id="showregistrations"></pre>
```
Comment (fabianfrz) [inline net/siproxd/src/opnsense/mvc/app/views/OPNsense/Siproxd/general.volt]: broken: cannot have the same id

File: net/siproxd/src/opnsense/mvc/app/views/OPNsense/Siproxd/general.volt [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+                    <th data-column-id="port" data-type="string" data-visible="true">{{ lang._('Port') }}</th>
+                </tr>
+            </thead>
+            <tbody>
+            </tbody>
+            <tfoot>
+                <tr>
+                    <td colspan="5"></td>
+                <td>
+                    <button data-action="add" type="button" class="btn btn-xs btn-default"><span class="fa fa-plus"></span></button>
```
Comment (fabianfrz) [inline net/siproxd/src/opnsense/mvc/app/views/OPNsense/Siproxd/general.volt]: this two lines are missing an indent step

File: net/siproxd/src/opnsense/mvc/app/views/OPNsense/Siproxd/general.volt [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+                    <th data-column-id="password" data-type="string" data-visible="true">{{ lang._('Password') }}</th>
+                </tr>
+            </thead>
+            <tbody>
+            </tbody>
+            <tfoot>
+                <tr>
+                    <td colspan="5"></td>
+                <td>
+                    <button data-action="add" type="button" class="btn btn-xs btn-default"><span class="fa fa-plus"></span></button>
```
Comment (fabianfrz) [inline net/siproxd/src/opnsense/mvc/app/views/OPNsense/Siproxd/general.volt]: this two lines are missing an indent step

File: net/siproxd/src/opnsense/mvc/app/models/OPNsense/Siproxd/User.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
+<?php
+namespace OPNsense\Siproxd;
+
+use OPNsense\Base\BaseModel;
```
Comment (fabianfrz) [inline net/siproxd/src/opnsense/mvc/app/models/OPNsense/Siproxd/User.php]: I can remember that @fichtner asked me to move that after the license header some time ago.

File: net/siproxd/src/opnsense/mvc/app/models/OPNsense/Siproxd/General.xml [hunk 6]
```diff
@@ -0,0 +1,182 @@
...
+        </ua_string>
+        <use_rport type="OptionField">
+            <default>Option1</default>
+            <multiple>N</multiple>
+            <Required>Y</Required>
+                <OptionValues>
+                    <Option1 value="0">0 - do not add ;rport to via header</Option1>
+                    <Option2 value="1">1 - do add ;rport to INCOMING via header only</Option2>
+                    <Option3 value="2">2 - do add ;rport to OUTGOING via header only</Option3>
+                    <Option4 value="3">3 - do add ;rport to OUTGOING and INCOMING via headers</Option4>
```
Comment (fabianfrz) [inline net/siproxd/src/opnsense/mvc/app/models/OPNsense/Siproxd/General.xml]: afaik threre is no value attribute. the tag is the value and the string inbetween is the visible string

File: net/siproxd/src/opnsense/mvc/app/models/OPNsense/Siproxd/Domain.php [hunk 7]
```diff
@@ -0,0 +1,30 @@
+<?php
+namespace OPNsense\Siproxd;
+
+use OPNsense\Base\BaseModel;
```
Comment (fabianfrz) [inline net/siproxd/src/opnsense/mvc/app/models/OPNsense/Siproxd/Domain.php]: see comment later

File: net/siproxd/src/opnsense/mvc/app/controllers/OPNsense/Siproxd/GeneralController.php [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+
+namespace OPNsense\Siproxd;
+
+class GeneralController extends \OPNsense\Base\IndexController
+{
+    public function indexAction()
+    {
+        $this->view->title = gettext("Siproxd settings");
```
Comment (fabianfrz) [inline net/siproxd/src/opnsense/mvc/app/controllers/OPNsense/Siproxd/GeneralController.php]: not an requirement but maybe upcase the S in Settings?

File: net/siproxd/src/opnsense/mvc/app/controllers/OPNsense/Siproxd/Api/ServiceController.php [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+
+            $runStatus = $this->statusAction();
+
+            // stop siproxd if it is running or not
+            $this->stopAction();
+
+            // generate template
+            $backend->configdRun('template reload OPNsense/Siproxd');
+
+            // (res)start daemon
```
Comment (fabianfrz) [inline net/siproxd/src/opnsense/mvc/app/controllers/OPNsense/Siproxd/Api/ServiceController.php]: you copied my typo ;)

**Comment by fichtner** [comment]
you can use &lt;textarea&gt; or &lt;pre&gt; for monospace output

File: net/siproxd/src/opnsense/mvc/app/models/OPNsense/Siproxd/General.xml (hunk 6 above)
Comment (mimugmail) [reply net/siproxd/src/opnsense/mvc/app/models/OPNsense/Siproxd/General.xml]: Yes, this requires patching:
https://github.com/opnsense/core/commit/c7c1c55c8e4c2e72c84ebd6e9212c4faa25ef9ee#commitcomment-24437550

File: net/siproxd/src/opnsense/mvc/app/controllers/OPNsense/Siproxd/Api/ServiceController.php (hunk 9 above)
Comment (mimugmail) [reply net/siproxd/src/opnsense/mvc/app/controllers/OPNsense/Siproxd/Api/ServiceController.php]: best copy+paster ever :) Fixed

File: net/siproxd/src/opnsense/mvc/app/controllers/OPNsense/Siproxd/GeneralController.php (hunk 8 above)
Comment (mimugmail) [reply net/siproxd/src/opnsense/mvc/app/controllers/OPNsense/Siproxd/GeneralController.php]: Hm, I'm cloning always from the latest plugins, so far it was never requested to change it to upper case. Doing a fgrep in plugins results in a mix of upper and lower.

File: net/siproxd/src/opnsense/mvc/app/models/OPNsense/Siproxd/ACL/ACL.xml (hunk 1 above)
Comment (mimugmail) [reply net/siproxd/src/opnsense/mvc/app/models/OPNsense/Siproxd/ACL/ACL.xml]: Wow, I really missed that file :/ Fixed

File: net/siproxd/src/opnsense/mvc/app/views/OPNsense/Siproxd/general.volt (hunk 4 above)
Comment (mimugmail) [reply net/siproxd/src/opnsense/mvc/app/views/OPNsense/Siproxd/general.volt]: Fixed

File: net/siproxd/src/opnsense/mvc/app/views/OPNsense/Siproxd/general.volt (hunk 3 above)
Comment (mimugmail) [reply net/siproxd/src/opnsense/mvc/app/views/OPNsense/Siproxd/general.volt]: fixed

File: net/siproxd/src/opnsense/mvc/app/views/OPNsense/Siproxd/general.volt (hunk 2 above)
Comment (mimugmail) [reply net/siproxd/src/opnsense/mvc/app/views/OPNsense/Siproxd/general.volt]: Yay .. this was also responsible for the broken output! Looks good now, thanks!

File: net/siproxd/src/opnsense/mvc/app/controllers/OPNsense/Siproxd/GeneralController.php (hunk 8 above)
Comment (fabianfrz) [reply net/siproxd/src/opnsense/mvc/app/controllers/OPNsense/Siproxd/GeneralController.php]: the command `git grep "view->title" | cut -d "=" -f 2` tells something else...

spectacle nn3815

Comment (fabianfrz) [reply net/siproxd/src/opnsense/mvc/app/controllers/OPNsense/Siproxd/GeneralController.php]: @fraenki the comand reveals that your controllers are missing `gettext` calls for translation.

Comment (mimugmail) [reply net/siproxd/src/opnsense/mvc/app/controllers/OPNsense/Siproxd/GeneralController.php]: I just searched in plugins, there it's quite balanced:

$this->view->title = gettext("FreeRADIUS Settings");
<?=gettext("L2TP settings"); ?></b></td>
<?=gettext("PPTP settings"); ?></b></td>
$this->view->title = gettext("BGP Settings");
$this->view->title = gettext("Routing-Settings");
$this->view->title = gettext("IS-IS-Settings");
$this->view->title = gettext("OSPFv3 Settings");
$this->view->title = gettext("OSPF Settings");
$this->view->title = gettext("RIP Settings");
<?=gettext("Global settings"); ?></strong>
<?=gettext("UPnP and NAT-PMP Settings");?></strong>
<?= gettext('This service can be used to wake up (power on) computers by sending special "Magic Packets". The NIC in the computer that is to be woken up must support Wake on LAN and has to be configured properly (WOL cable, BIOS settings).');?>
$this->view->title = gettext("Siproxd settings");
$this->view->title = gettext("Siproxd settings");
$this->view->title = gettext("Collectd settings");
$this->view->title = gettext('Zabbix Agent Settings');
$this->view->title = gettext("ClamAV settings");
$this->view->title = gettext('Monit System Monitoring - Settings');
$this->view->title = gettext("C-ICAP Settings");

Comment (fabianfrz) [reply net/siproxd/src/opnsense/mvc/app/controllers/OPNsense/Siproxd/GeneralController.php]: @mimugmail for me this looks like there is a difference between the MVC plugins and the legacy plugins.

Maybe @fichtner or @AdSchellevis has a coding convention here.

Comment (fichtner) [reply net/siproxd/src/opnsense/mvc/app/controllers/OPNsense/Siproxd/GeneralController.php]: Uppercase for titles is preferred but I wouldn't bikeshed over this :)

Comment (mimugmail) [reply net/siproxd/src/opnsense/mvc/app/controllers/OPNsense/Siproxd/GeneralController.php]: Ok, I'll try to make the others upper case too when this is through :)

File: net/siproxd/src/opnsense/mvc/app/models/OPNsense/Siproxd/User.php (hunk 5 above)
Comment (mimugmail) [reply net/siproxd/src/opnsense/mvc/app/models/OPNsense/Siproxd/User.php]: fixed

File: net/siproxd/src/opnsense/mvc/app/models/OPNsense/Siproxd/Domain.php (hunk 7 above)
Comment (mimugmail) [reply net/siproxd/src/opnsense/mvc/app/models/OPNsense/Siproxd/Domain.php]: fixed

**Comment by mimugmail** [comment]
I removed the WIP, if you don't find anything else we could merge and let @NicholasRush test the whole thing.

File: net/siproxd/Makefile [hunk 10]
```diff
@@ -0,0 +1,8 @@
+PLUGIN_NAME=		siproxd
+PLUGIN_VERSION=		0.1
+PLUGIN_COMMENT=		Siproxd is a proxy daemon for the SIP protocol
```
Comment (fabianfrz) [inline net/siproxd/Makefile]: Maybe shorten this to "Proxy daemon for the SIP protocol"

Comment (fichtner) [reply net/siproxd/Makefile]: SIP means session initiation protocol :P

File: net/siproxd/pkg-descr [hunk 11]
```diff
@@ -0,0 +1,10 @@
+Siproxd is a proxy/masquerading daemon for the SIP protocol.
+It handles registrations of SIP clients on a private IP network
+and performs rewriting of the SIP message bodies to make SIP
+connections work via an masquerading firewall (NAT).
+It allows SIP software clients (like kphone, linphone) or
+SIP hardware clients (Voice over IP phones which are
+SIP-compatible, such as those from Cisco, Grandstream or Snom)
+to work behind an IP masquerading firewall or NAT router.
+
+WWW: http://siproxd.sourceforge.net/
```
Comment (fichtner) [inline net/siproxd/pkg-descr]: We can set PLUGIN_WWW from the makefile for this if needed, we don't want it in the description (a FreeBSD ports artifact as that gets scraped from the file as the WWW entry as well).

File: net/siproxd/src/opnsense/scripts/OPNsense/Siproxd/setup.sh [hunk 12]
```diff
@@ -0,0 +1,9 @@
+#!/bin/sh
+
+mkdir -p /var/run/siproxd
+chown -R nobody:nogroup /var/run/siproxd
+chmod 750 /var/run/siproxd
+
+mkdir -p /var/lib/siproxd
+chown -R nobody:nogroup /var/lib/siproxd
+chmod 750 /var/lib/siproxd
```
Comment (fichtner) [inline net/siproxd/src/opnsense/scripts/OPNsense/Siproxd/setup.sh]: nice work pinning these. freebsd does not seem to list them: https://github.com/opnsense/ports/blob/47aa039c8c8f361193e18a636760392971ef57ae/net/siproxd/pkg-plist

File: net/siproxd/src/opnsense/service/conf/actions.d/actions_siproxd.conf [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+message:reconfigure Siproxd
+
+[status]
+command:/usr/local/etc/rc.d/siproxd status;exit 0
+parameters:
+type:script_output
+message:request Siproxd status
+
+[show-registrations]
+command:/bin/cat /var/lib/siproxd/siproxd_registrations
```
Comment (fichtner) [inline net/siproxd/src/opnsense/service/conf/actions.d/actions_siproxd.conf]: does this require /var persistency in setup.sh?

**Comment by fichtner** [comment]
Merged, have some time, will fix my comments in the tree
